src/zenml/zen_service/zen_service_api.py

```diff
@@ -0,0 +1,186 @@
+        ) from error
+
+
+@app.get(
+    "/stacks/get-configurations",
+    response_model=Dict[str, Dict[StackComponentType, str]],
+)
+async def stack_configurations() -> Dict[str, Dict[StackComponentType, str]]:
+    return stack_store.stack_configurations
+
+
+@app.post("/components/register", responses={409: error_response})
+async def register_stack_component(
+    component: StackComponentWrapper,
+) -> None:
+    try:
+        stack_store.register_stack_component(component)
+    except StackComponentExistsError as error:
+        raise HTTPException(
+            status_code=status.HTTP_409_CONFLICT, detail=error.args
+        ) from error
+
+
+@app.get("/stacks/delete/{name}")
+async def deregister_stack(name: str) -> None:
+    logging.warning(f"About to delete stack {name}")
+    try:
+        stack_store.deregister_stack(name)
+    except KeyError as error:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND, detail=error.args
+        ) from error
+
+
+@app.get("/stacks/get", response_model=List[StackWrapper])
+async def stacks() -> List[StackWrapper]:
+    return stack_store.stacks
+
+
+@app.get(
+    "/stacks/get/{name}",
+    response_model=StackWrapper,
+    responses={404: error_response},
+)
+async def get_stack(name: str) -> StackWrapper:
+    try:
+        return stack_store.get_stack(name)
+    except KeyError as error:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND, detail=error.args
+        ) from error
+
+
+@app.post(
+    "/stacks/register",
+    response_model=Dict[str, str],
+    responses={404: error_response, 409: error_response},
+)
+async def register_stack(stack: StackWrapper) -> Dict[str, str]:
+    try:
+        return stack_store.register_stack(stack)
+    except StackComponentExistsError as error:
+        raise HTTPException(
+            status_code=status.HTTP_409_CONFLICT, detail=error.args
+        ) from error
+    except StackExistsError as error:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND, detail=error.args
+        ) from error
+
+
+@app.get(
+    "/components/get/{component_type}/{name}",
+    response_model=StackComponentWrapper,
+)
+async def get_stack_component(
+    component_type: StackComponentType, name: str
+) -> StackComponentWrapper:
+    return stack_store.get_stack_component(component_type, name=name)
+
+
+@app.get(
+    "/components/get/{component_type}",
+    response_model=List[StackComponentWrapper],
+)
+async def get_stack_components(
+    component_type: StackComponentType,
+) -> List[StackComponentWrapper]:
+    return stack_store.get_stack_components(component_type)
+
+
+@app.get("/components/delete/{component_type}/{name}")
+async def deregister_stack_component(
+    component_type: StackComponentType, name: str
+) -> None:
+    try:
+        return stack_store.deregister_stack_component(component_type, name=name)
```

Comment:
Now that deregistering stacks was updated to raise a `KeyError` (and return a 404 response if a stack with that name doesn't exist), should we handle stack components in a similar way?